src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1426,15 +1430,54 @@ private IEnumerable<ITypeSymbol> InferTypeInNameColon(NameColonSyntax nameColon,
                 return SpecializedCollections.EmptyEnumerable<ITypeSymbol>();
             }
 
-            private IEnumerable<ITypeSymbol> InferTypeInMemberAccessExpression(MemberAccessExpressionSyntax expression)
+            private IEnumerable<ITypeSymbol> InferTypeInMemberAccessExpression(
+                MemberAccessExpressionSyntax memberAccessExpression,
+                ExpressionSyntax expressionOpt = null, 
```

Comment:
So the pattern for type inference is that we walk upwards finding a proper context that we can determine a type for.  The node we're currently on is the first arg.  If we walked up from a cihld, then that's the second arg.  If this was a positional inference, then this is the token on the left of the cursor.  

Inference is used for two purposes.  One is when we have an actual node and we're tring to decide what type it should have.  i.e. "int i = [|Foo()|]".  Here we want to say "Foo() should have type 'int'".

The second is when we're just typing and we want to know what type we would infer to drive features like completion.  for example if i have "Color c = $$", then i want to say "i infer Color here" so we can do enum preselection.

This pattern should be consistent throughout the entire feature for C# and VB.